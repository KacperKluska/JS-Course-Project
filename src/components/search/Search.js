import "./style.scss";

export default function Search() {
  return (
    <div id="search">
      <input type="text" id="searchInput" placeholder="Search..." />
    </div>
  );
}
