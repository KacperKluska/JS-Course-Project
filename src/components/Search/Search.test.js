import Search from './Search';

let mountWrapper;

beforeAll(() => {
  mountWrapper = mount(<Search class="active" />);
});

test('simulate click', () => {
  const spyFunction = jest.fn();
  const newWrapper = mount(
    <Search searchRef={null} class="active" onClick={spyFunction} />,
  );

  expect(spyFunction).toBeCalledTimes(0);
  newWrapper.find('.navItemButton').simulate('click');
  expect(spyFunction).toBeCalledTimes(1);
});

test('Search snapshot, different snap test', () => {
  const container = renderer.create(<Search class="active" />);
  const shallowWrapper = shallow(<Search class="active" />);
  const renderWrapper = render(<Search class="active" />);
  expect(container.toJSON()).toMatchSnapshot();
  expect(mountWrapper).toMatchSnapshot();
  expect(shallowWrapper).toMatchSnapshot();
  expect(renderWrapper).toMatchSnapshot();
});
