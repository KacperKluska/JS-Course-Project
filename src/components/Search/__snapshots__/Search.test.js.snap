// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search snapshot, different snap test 1`] = `
<div
  id="search"
>
  <li
    className="navItem"
  >
    <button
      className="navItemButton false"
      disabled={false}
      type="button"
    >
      <i
        className="icon-cancel"
      />
      Close
    </button>
  </li>
  <input
    id="searchInput"
    placeholder="Search..."
    type="text"
  />
</div>
`;

exports[`Search snapshot, different snap test 2`] = `
<div
  id="search"
>
  <li
    className="navItem"
  >
    <button
      className="navItemButton false"
      disabled={false}
      type="button"
    >
      <i
        className="icon-cancel"
      />
      Close
    </button>
  </li>
  <input
    id="searchInput"
    placeholder="Search..."
    type="text"
  />
</div>
`;

exports[`Search snapshot, different snap test 3`] = `
<div
  id="search"
>
  <NavItem
    icon="icon-cancel"
  >
    Close
  </NavItem>
  <input
    id="searchInput"
    placeholder="Search..."
    type="text"
  />
</div>
`;

exports[`Search snapshot, different snap test 4`] = `
<div
  id="search"
>
  <li
    class="navItem"
  >
    <button
      class="navItemButton false"
      type="button"
    >
      <i
        class="icon-cancel"
      />
      Close
    </button>
  </li>
  <input
    id="searchInput"
    placeholder="Search..."
    type="text"
  />
</div>
`;
